### DISCUSSION TITLE
is there a way to add more metadata (like uuid) for path operation itself?

### DISCUSSION THREAD

**Discussion by shizidushu** [fastapi/fastapi#8957] [Questions]
I want to assign an unique identifier for each path operation of a router. Then I can collect the `app.openapi()` data and put it into database for further use.

`operationid`, `path`, `summary`, `description` of a path operation can be changed easily and it is not clear to be unique and unchangeable.

I wish there is something like
```python
@router.get(
    "/get_orders",
    uuid = "2dd94d0d-0862-4a04-ad54-ac478ac4b7e9"
)
```
Obviously, the above code doesn't work because there is no `uuid` argument.

Is there is a way to get around?

**Comment by ycd** [comment]
You can do something like this, which will create you a unique operationID based on your function name.

```python
from fastapi import FastAPI
from fastapi.routing import APIRoute

app = FastAPI()


@app.get("/items/")
async def read_items():
    return [{"item_id": "Foo"}]


def use_route_names_as_operation_ids(app: FastAPI) -> None:
    """
    Simplify operation IDs so that generated API clients have simpler function
    names.

    Should be called only after all routes have been added.
    """
    for route in app.routes:
        if isinstance(route, APIRoute):
            route.operation_id = route.name  # in this case, 'read_items'


use_route_names_as_operation_ids(app)
```

Reference: [Using the path operation function name as the operationId](https://fastapi.tiangolo.com/advanced/path-operation-advanced-configuration/)

**Comment by shizidushu** [comment]
@ycd Thank you for your reply. Function name can be changed easily too. I like `uuid` if I can specify because it gives others (includes future me) a strong hint that there is no need to change it.

**Comment by shizidushu** [comment]
Just tried to use function name as operationId. Fastapi seems doesn't enforce the `operationId` to be unique.

**Comment by septatrix** [comment]
Regarding your initial question the automatically assigned `operationId` will always be unique as it combines function name, endpoint path and HTTP method into the id which in combination _should_ never be the same (though FastAPI does not enforce this)

> Just tried to use function name as operationId. Fastapi seems doesn't enforce the `operationId` to be unique.

Personally I would like if FastAPI would enforce unique operationIds (and unique method-path-name tuples) when manually assigning them however as you observed it currently does not.
